test/powershell/Host/PSVersionTable.Tests.ps1

```diff
@@ -15,20 +40,31 @@ Describe "PSVersionTable" -Tags "CI" {
        $PSVersionTable.ContainsKey("OS")                        | Should Be True
 
     }
-    It "GitCommitId property should not contain an error" {
-       $PSVersionTable.GitCommitId | Should not match "powershell.version"
+
+    It "PSVersion property" {
+       $PSVersionTable.PSVersion | Should BeOfType "System.Management.Automation.SemanticVersion"
+       $PSVersionTable.PSVersion | Should BeExactly $expectedPSVersion
+       $PSVersionTable.PSVersion | Should Match $expectedVersionPattern
+       $PSVersionTable.PSVersion.Major | Should Be 6
+    }
+
+    It "GitCommitId property" {
+       $PSVersionTable.GitCommitId | Should BeOfType "System.String"
+       $PSVersionTable.GitCommitId | Should Match $expectedGitCommitIdPattern
+       $PSVersionTable.GitCommitId | Should Not Match $unexpectectGitCommitIdPattern
```

Comment:
This `not match` test case is not necessary IMO since you are doing a `BeExactly` comparison below.

Reply:
We do two checks - 1. for string values (line below). 2. for regex patterns (this line).
I split regex patterns to check GitCommitId and ReleaseTag cases and keep the patterns simple. The double "Should" is needed because `ReleaseTag` pattern can pass `GitCommitId`.

Reply:
I see the point. It's to guarantee that the reconstructed `rawGitCommitId` to be in the expected regx pattern. #Close